splice_machine/src/test/java/com/splicemachine/benchmark/FeatureStoreTriggerBenchmark.java

```diff
@@ -200,23 +184,45 @@ private static void populateStagingTable(long ts0, long ts1) {
     }
 
     private void populateStagingTable() throws SQLException {
-        testStatement.execute("TRUNCATE TABLE FeatureStaging");
-        long nextTimestamp = curTimestamp + 60 * DAYMS;
+        testStatement.execute("DROP TABLE IF EXISTS FeatureStaging");
+        String fileName = "/tmp/split_FeatureStaging";
+        testStatement.execute("DROP TABLE IF EXISTS SPLITKEYS");
+        testStatement.execute("CREATE TABLE SPLITKEYS (pk1 int, pk2 int)");
+        for (int split = 1; split < numHosts; ++split) {
+            int pk = split * (entities / numHosts);
+            testStatement.execute("INSERT INTO SPLITKEYS VALUES (" + pk + ",0)");
+        }
+        testStatement.execute(String.format("EXPORT('%s', false, null, null, null, null) SELECT * FROM SPLITKEYS", fileName));
+
+        testStatement.execute(
+                String.format("CREATE TABLE FeatureStaging(" +
+                        "entity_key INTEGER," +
+                        "update_order INTEGER," +
+                        "ts TIMESTAMP," +
+                        "feature1 DOUBLE," +
+                        "feature2 DOUBLE," +
+                        "PRIMARY KEY (entity_key, update_order))" +
+                        "LOGICAL SPLITKEYS LOCATION '%s'", fileName)
+        );
+
+        long nextTimestamp = curTimestamp + updates * 2 * DAYMS;
         taskId.set(0);
         runBenchmark(connections, () -> populateStagingTable(curTimestamp, nextTimestamp));
         curTimestamp = nextTimestamp;
+
+        testStatement.execute("ANALYZE TABLE FeatureStaging");
```

Comment:
I don't think this would actually help Spark balance tasks better, if anything just because we are delaying the execution and HBase might complete flushes/compactions/splits, but the analyze itself wouldn't play a role I think.

If you wanted that you could force a flush and major_compaction, I think that would help more.

Reply:
I think you are right. I was able to get comparable performance with a simple delay instead of ANALYZE.
FLUSH + MAJOR COMPACTION improve batch performance from 35K op/sec to 50K ops/sec (but only in one out of the two tests).